src/lib/external/curve/router/call.ts

```diff
@@ -0,0 +1,176 @@
+import { IRoute } from "@curvefi/api/lib/interfaces";
+import { Hash, encodeAbiParameters, parseAbiParameters } from "viem";
+
+const EMPTY_ROUTE = [ADDRESS_ZERO, ADDRESS_ZERO, ADDRESS_ZERO, ADDRESS_ZERO, ADDRESS_ZERO, ADDRESS_ZERO, ADDRESS_ZERO, ADDRESS_ZERO, ADDRESS_ZERO]
+
+const EMPTY_SWAP_PARAMS = [
+    [ZERO, ZERO, ZERO],
+    [ZERO, ZERO, ZERO],
+    [ZERO, ZERO, ZERO],
+    [ZERO, ZERO, ZERO]
+]
+
+const curveInit: () => Promise<void> = async () => {
+    await curve.init("Alchemy", { network: "homestead", apiKey: process.env.NEXT_PUBLIC_ALCHEMY_API_KEY as string }, { chainId: 1 });
+    await curve.factory.fetchPools();
+    await curve.crvUSDFactory.fetchPools();
+    await curve.EYWAFactory.fetchPools();
+    await curve.cryptoFactory.fetchPools();
+    await curve.tricryptoFactory.fetchPools();
+}
+
+function processRoute(route: IRoute): CurveRoute {
+    const routeSteps: string[] = [];
+    const swapParams: BigInt[][] = [];
+    for (let i = 0; i < route.length; i++) {
+        swapParams[i] = [BigInt(route[i].swapParams[0]), BigInt(route[i].swapParams[1]), BigInt(route[i].swapParams[2])];
+        if (i === 0) {
+            routeSteps.push(route[i].inputCoinAddress);
+            routeSteps.push(route[i].poolAddress);
+            routeSteps.push(route[i].outputCoinAddress);
+        } else {
+            routeSteps.push(route[i].poolAddress);
+            routeSteps.push(route[i].outputCoinAddress);
+        }
+    }
+    if (routeSteps.length < 9) {
+        const addZeroAddresses = 9 - routeSteps.length;
+        for (let i = 0; i < addZeroAddresses; i++) {
+            routeSteps.push(ADDRESS_ZERO)
+        }
+    }
+    if (swapParams.length < 4) {
+        const addEmptySwapParams = 4 - swapParams.length;
+        for (let i = 0; i < addEmptySwapParams; i++) {
+            swapParams.push([ZERO, ZERO, ZERO]);
+        }
+    }
+
+    const curveRoute: CurveRoute = {
+        route: routeSteps,
+        swapParams: swapParams
+    }
+    return curveRoute;
+}
+
+
+export const curveApiCall = async ({
+    depositAsset,
+    rewardTokens,
+    baseAsset,
+    router,
+    minTradeAmounts,
+    optionalData
+}: {
+    depositAsset: string,
+    rewardTokens: string[],
+    baseAsset: string,
+    router: string,
+    minTradeAmounts: BigInt[],
+    optionalData: Hash
+}): Promise<{ baseAsset: string, router: string, toBaseAssetRoutes: CurveRoute[], toAssetRoute: CurveRoute, minTradeAmounts: BigInt[], optionalData: Hash }> => {
+    if (rewardTokens.length !== minTradeAmounts.length) {
+        throw new Error("rewardTokens and minTradeAmounts must be the same length");
+    }
+
+    await curveInit();
+
+    const toBaseAssetRoutes: CurveRoute[] = [];
+    for (let i = 0; i < rewardTokens.length; i++) {
+        const inputToken = rewardTokens[i];
+        const outputToken = baseAsset;
+        const { route: rewardRoute, } = await curve.router.getBestRouteAndOutput(inputToken, outputToken, '100000000');
```

Comment:
what's that number used for? Can we store it in a variable to give it a descriptive name?